src/zelos/plugin/syscall_manager_base.py

```diff
@@ -185,10 +188,13 @@ def handle_syscall(self, process):
             # The current thread might get modified by the syscall.
             thread = self.z.current_thread
             self.last_syscall_args = None
+            self._current_syscall = sys_name
             retval = sys_fn(self, process)
+            self._current_syscall = None
             if retval is not None:
                 self.set_return_value(retval)
         except Exception as e:
+            self._current_syscall = None
```

Comment:
Maybe include the current_syscall in the error logging? It's worth knowing if an error occurred inside the syscall impl

Reply:
I'll move the rest of the code in the exception out, so exceptions that are caught are very likely the syscall implementation